Reword SD-JWT credential validity text + remove requirement for verifiers/wallets to support status list (#276)
vcstuff/oid4vc-haip-sd-jwt-vc: openid4vc-high-assurance-interoperability-profile-1_0.md

This profile defines the following additional requirements for IETF SD-JWT VCs as defined in [@!I-D.ietf-oauth-sd-jwt-vc].

* Compact serialization MUST be supported as defined in [@!I-D.ietf-oauth-selective-disclosure-jwt]. JSON serialization MAY be supported.

> * If the Issuer wishes to limit the validity period of an SD-JWT VC it MUST use an `exp` claim, or a `status` claim, or both. The Wallet and the Verifier MUST support both mechanisms.

Author: I've added a recommendation.

> * It is RECOMMENDED that Issuers limit the validity period when issuing SD-JWT VC. When doing so, the Issuer MUST use an `exp` claim, a `status` claim, or both. The Wallet and the Verifier MUST support both mechanisms.

Reviewer: ```suggestion
* It is RECOMMENDED that Issuers limit the validity period when issuing SD-JWT VC. When doing so, the Issuer MUST use an `exp` claim, a `status` claim, or both.
```

> * It is RECOMMENDED that Issuers limit the validity period when issuing SD-JWT VC. If the Issuer wishes to do so, it MUST use an `exp` claim, a `status` claim, or both. The Wallet and the Verifier MUST support both mechanisms.

Reviewer 2: This might be playing semantics, but I find that "wish" further weakens the recommendation to do so. Maybe keep it cleaner as follows:

```suggestion
* It is RECOMMENDED that Issuers limit the validity period when issuing SD-JWT VC. When doing so, the Issuer MUST use an `exp` claim, a `status` claim, or both. The Wallet and the Verifier MUST support both mechanisms.
```

This profile defines the following additional requirements for IETF SD-JWT VCs as defined in [@!I-D.ietf-oauth-sd-jwt-vc].

* Compact serialization MUST be supported as defined in [@!I-D.ietf-oauth-selective-disclosure-jwt]. JSON serialization MAY be supported.
* It is RECOMMENDED that Issuers limit the validity period when issuing SD-JWT VC. When doing so, the Issuer MUST use an `exp` claim, a `status` claim, or both.
* The `cnf` claim [@!RFC7800] MUST conform to the definition given in [@!I-D.ietf-oauth-sd-jwt-vc]. Implementations conforming to this profile MUST include the JSON Web Key [@!RFC7517] in the `jwk` member if the corresponding Credential Configuration requires cryptographic holder binding.

> * The `status` claim, if present, MUST use the Token Status List as defined in [@!I-D.ietf-oauth-status-list]

Reviewer: ```suggestion
* The `status` claim, if present, MUST use the Token Status List in JWT format as defined in [@!I-D.ietf-oauth-status-list]
```

Author: ```suggestion
* The `status` claim, if present, MUST contain `status_list` as defined in [@!I-D.ietf-oauth-status-list]
```
